frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/StartPipelineModal.tsx

```diff
@@ -31,11 +34,28 @@ const StartPipelineModal: React.FC<StartPipelineModalProps & ModalComponentProps
       ...workspace,
       type: 'EmptyDirectory',
     })),
-    secretOpen: false,
+    newSecrets: [],
+    secretForm: secretFormDefaultValues,
   };
 
   const handleSubmit = (values: StartPipelineFormValues, actions): void => {
     actions.setSubmitting(true);
+    const createSecrets = [];
+    values.newSecrets.map((data) => {
+      const secret = k8sCreate(SecretModel, data);
+      createSecrets.push(secret);
+      return createSecrets;
+    });
```

Comment:
Map returns the array of values you returned on each pass... you can just assign it.
```suggestion
    const createSecrets = values.newSecrets.map((data) => {
      return k8sCreate(SecretModel, data);
    });
```